### PR TITLE
AS-789 Post test results to slack when code is pushed to dev

### PR DESCRIPTION
**Pull request by ungwudik**
PR for [AS-789](https://broadworkbench.atlassian.net/browse/AS-789), following the discussion in this [doc](https://docs.google.com/document/d/1_rHOGC4bvbdTusJbCVurqCs6Jn9YpYLHTa7DFnnodqw/edit?usp=sharing)

- This PR handles one part (i.e. post test results to slack when code is pushed to ~~dev~~ main branch)
- Another small/accompanying [PR in dsp-jenkins repo](https://github.com/broadinstitute/dsp-jenkins/pull/538) handles the other parts
   - Remove WSM Unit and Connected tests as part of release promotion runs in alpha and staging
   - Don't kick off WSM tests after deploy to dev (doesn't provide any additional coverage).

Sample screenshots of what Slack posts will look like
Screen Shot 2021-08-12 at 3 24 15 PM

### CODE REVIEW

File: .github/workflows/pr-integration.yml [hunk 1]
```diff
@@ -93,3 +93,17 @@ jobs:
       with:
         test-server: workspace-local.json
         test: suites/FullIntegration.json
+
+    - name: "Notify WSM Slack"
+      if: always() && github.event_name == 'push' && github.ref == 'refs/heads/main'
```
Comment (ungwudik) [inline .github/workflows/pr-integration.yml:98]: I have not tested with this condition here (i.e. `if: always() && github.event_name == '**push**' && github.ref == '**refs/heads/main**'`) as that requires…a push to main branch.

I tested against this very PR using `if: always() && github.event_name == '**pull_request**' && github.ref == '**refs/pull/421/merge**'`. I'll ensure to follow up after merging this PR to main to make sure everything looks good.

Comment (MatthewBemis) [reply .github/workflows/pr-integration.yml:98]: I'm not too well-versed in GHA, but why is the `always()` needed at the beginning of the conditional?

(same question in test.yml)

**Review by MatthewBemis — APPROVED** [review]

File: .github/workflows/pr-integration.yml (hunk 1 above)
Comment (ungwudik) [reply .github/workflows/pr-integration.yml:98]: The `always()` tells GHA to run the job no matter what the status (success, failure) is. But yeah, the `always()` function is a bit odd here, especially when then combined with `&&`.

This short, related discussion from [here](https://github.com/DataBiosphere/terra-workspace-manager/pull/378#discussion_r649429130) from earlier may also help.

**Comment by ansingh7115** [comment]
I would change the title of this PR to be something more descriptive

File: .github/workflows/pr-integration.yml (hunk 1 above)
Comment (ansingh7115) [reply .github/workflows/pr-integration.yml:98]: what's the difference between the behavior of these two:
`if: always() && github.event_name == 'push' && github.ref == 'refs/heads/main'`
and 
`if: github.event_name == 'push' && github.ref == 'refs/heads/main'`

Comment (ungwudik) [reply .github/workflows/pr-integration.yml:98]: In `if: always() && github.event_name == 'push' && github.ref == 'refs/heads/main'`, results will post to Slack no matter the status of the previous steps (success, failure, cancelled).

In `if: github.event_name == 'push' && github.ref == 'refs/heads/main'`, GitHub will apply the default status check `success()`. Results will post to Slack only when none of the previous steps failed OR was cancelled.

https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions

Comment (ansingh7115) [reply .github/workflows/pr-integration.yml:98]: Awesome! Thanks - I think Matt and I asked the same question at the same time - I didn't realize he had left the same comment until I refreshed :)

**Review by ansingh7115 — APPROVED** [review]

File: .github/workflows/pr-integration.yml (hunk 1 above)
Comment (ungwudik) [reply .github/workflows/pr-integration.yml:98]: lol no worries

**Pull request merged**